streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -209,6 +210,7 @@ public StreamTask(final TaskId id,
 
         recordInfo = new PartitionGroup.RecordInfo();
         partitionGroup = new PartitionGroup(partitionQueues);
+        processorContextImpl.setStreamTimeSupplier(partitionGroup::timestamp);
```

Comment:
We should consider caching the stream time in `PartitionGroup` class than re-computing it every time, since previously it is only called twice per `runOnce`, i.e. per a batch of records, but now it is called for each record processed in windowed aggregate.

Reply:
@guozhangwang I've done this in https://github.com/apache/kafka/pull/5218/commits/05175f4ec371ee8efbbe4d3fd4b2db9016135c6f

It's really doing the same amount of work, but moving it to poll-time to ensure we only pay the cost once per record.